docs/docs/preprocessing-external-images.md

```diff
@@ -0,0 +1,151 @@
+---
+title: Preprocessing External Images
+---
+
+Gatsby allows powerful image processing features through using the [`Sharp`](https://github.com/lovell/sharp/) library to automatically process images to be performant and have features like lazy-loading. That said, this only works if the image is a file node in the GraphQL layer.
+
+If you want the same ecosystem for files that are hosted online and not located in your repo, [`gatsby-source-filesystem`](/packages/gatsby-source-filesystem/) has an API called `createRemoteFileNode` to solve this.
+
+This guide will walk you through how to use `createRemoteFileNode` process and get the same benefits of gatsby-transformer-sharp with externally sourced images.
+
+## Setup
+
+A usecase that this technique can be useful is if you want to create a featured image in a blogpost, but the image is a url up on the web instead of a local file. This could be hosted somewhere like Imgur, S3, or anywhere up on the internet.
+
+Give a sample post:
+
+```markdown
+---
+title: My first blogpost!
+featuredImgUrl: https://images.unsplash.com/photo-1560237731-890b122a9b6c
+---
+
+Hello World
+```
+
+We can have a frontmatter field for the url of the featured image we want to pull down and use as part of the site.
+
+By default, this is just a string as we haven't told Gatsby how to interpret it, but now we can add some code into `gatsby-node.js` to modify it.
+
+## Gatsby Node
+
+In your gatsby-node file, we can do some processing to create file nodes for the `featuredImgUrl` frontmatter field.
+
+```js:title=gatsby-node.js
+const { createRemoteFileNode } = require("gatsby-source-filesystem")
+
+exports.onCreateNode = async ({
+  node,
+  actions: { createNode },
+  store,
+  cache,
+  createNodeId,
+}) => {
+  // For all MarkdownRemark nodes, call createRemoteFileNode
+  if (node.internal.type === "MarkdownRemark") {
```

Comment:
Should we include a check for the `featuredImgUrl`?

Reply:
yes good idea. let me go update that

Reply:
Done and added a createTypes call so the featuredImgUrl can be optional